metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy

```diff
@@ -160,27 +166,25 @@ class FilterExperimentData implements FilterExperimentDataInput {
      * @param entryProps which contains all properties along with the filenames
      * @return validation status as boolean
      */
-    boolean validFilenames(HashMap entryProps) {
-        boolean valid = false
+    protected void validFilenames(HashMap entryProps) {
 
         String filename = (String) entryProps.get("Filename")
-        List<String> filesPerSample = Arrays.asList(filename.split(","))
+        String code = entryProps.get("QBiC.Code") as String
+        String seq = entryProps.get("SequencingFacilityId") as String
 
-        if (filesPerSample == null || filesPerSample.empty) {
-            return valid
+        if (filename == null || filename.empty) {
+            throw new InvalidFileNameException("There are no files given for your sample with the QBiC.Code $code")
         }
 
-        filesPerSample.each { file ->
+        List<String> filesPerSample = Arrays.asList(filename.split(","))
 
-            String code = entryProps.get("QBiC.Code") as String
-            String seq = entryProps.get("SequencingFacilityId") as String
```

Comment:
NIT: more expressive variable name ;)

Reply:
You found them :D I corrected the naming in 107d231